### PR TITLE
VY: Make logout link configurable

### PR DESCRIPTION
**Pull request by jediandrey**
- [ ] go to /virtual-ymca and login into it
- [ ] find Sign Out link and click, ensure you've been logged out
- [ ] login under admin user, go to /admin/openy/virtual-ymca
- [ ] find 'Virtual Y Logout url' end edit it like this '/user/logout?destination=join'
- [ ] repeat steps for login and logout, ensure you've been redirected to /join page

**Basecamp:** https://3.basecamp.com/3608022/buckets/17045978/todos/3092004786

### CODE REVIEW

**Review by anpolimus — APPROVED** [review]

**Comment by anpolimus** [comment]
Code is ok, please rebuild app and fix merge conflict

File: openy_gated_content.module [hunk 1]
```diff
@@ -57,6 +57,7 @@ function openy_gated_content_preprocess_paragraph(&$variables) {
   }
   $plugin_instance = $identityProviderManager->createInstance($active_provider);
   $app_config = \Drupal::config('openy_gated_content.settings')->getOriginal();
+  $variables['#attached']['drupalSettings']['openy_gated_content']['logoutUrl'] = !empty($app_config['virtual_y_logout_url']) ? $app_config['virtual_y_logout_url'] : base_path() . 'user/logout';
```
Comment (hamrant) [inline openy_gated_content.module:60]: No need to add this to `drupalSettings`, config already exists in `$variables['app_settings']`.

Example how to get this setting in component:

```js
  computed: {
    ...mapGetters([
      'getAppSettings',
    ]),
  },

```

or

```js
    logoutUrl() {
      return this.$store.getters.getAppSettings.virtual_y_logout_url;
    },
```

or use `SettingsMixin` that already includes second example

File: js/gated-content/src/components/LogoutLink.vue [hunk 2]
```diff
@@ -1,5 +1,16 @@
...
   <span class="menu">
-    <a href="/user/logout">Sign out</a>
+    <a :href="logoutUrl">Sign out</a>
   </span>
 </template>
+
+<script>
+export default {
+  name: 'LogoutLink',
+  data() {
```
Comment (hamrant) [inline js/gated-content/src/components/LogoutLink.vue:10]: Please use `SettingsMixin`, example:

```js
import { SettingsMixin } from '@/mixins/SettingsMixin';

export default {
  name: 'BlogListing',
  mixins: [SettingsMixin],
  computed: {
    logoutUrl() {
      return this.config.virtual_y_logout_url;
    },
  },

```

File: config/install/openy_gated_content.settings.yml [hunk 3]
```diff
@@ -19,6 +19,7 @@ components:
     up_next_title: 'Next blog posts'
 virtual_y_url: /virtual-ymca
 virtual_y_login_url: /virtual-y-login
+virtual_y_logout_url: /user/logout
```
Comment (hamrant) [inline config/install/openy_gated_content.settings.yml:22]: We need a hook update that will set only this value for existing projects

Comment (hamrant) [reply config/install/openy_gated_content.settings.yml:22]: Please update `openy_gated_content.schema.yml` too

**Review by hamrant — CHANGES_REQUESTED** [review]

**Review by anpolimus — APPROVED** [review]

**Pull request merged**